server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -151,18 +155,30 @@ public static Query refExistsQuery(Reference ref, Context context) {
             if (fieldType != null && !fieldType.omitNorms()) {
                 return new FieldExistsQuery(field);
             }
-            if (ArrayType.unnest(ref.valueType()) instanceof ObjectType objType) {
+
+            if (ref.valueType() instanceof ObjectType objType) {
+                if (objType.innerTypes().isEmpty()) {
+                    return null;
+                }
                 BooleanQuery.Builder booleanQuery = new BooleanQuery.Builder();
                 for (var entry : objType.innerTypes().entrySet()) {
                     String childColumn = entry.getKey();
                     Reference childRef = context.getRef(ref.column().append(childColumn));
-                    Query refExistsQuery = refExistsQuery(childRef, context);
+                    Query refExistsQuery = refExistsQuery(childRef, context, true);
                     if (refExistsQuery == null) {
                         return null;
                     }
                     booleanQuery.add(refExistsQuery, Occur.SHOULD);
                 }
-                return new ConstantScoreQuery(booleanQuery.build());
+                // Even if a child columns exist, an object can have empty values, we have to run generic function.
+                // Example of such object:
+                // CREATE TABLE t (obj OBJECT as (x int));
+                // INSERT INTO t (obj) VALUES ({});
+                return new BooleanQuery.Builder()
+                    .setMinimumNumberShouldMatch(1)
+                    .add(new ConstantScoreQuery(booleanQuery.build()), Occur.SHOULD)
```

Comment:
inspired by inspired by https://github.com/crate/crate/pull/12704/files#r913649625

can we/should we make left query a `BooleanQuery` and don't wrap it into `ConstantScoreQuery`?